### PR TITLE
Fix EZP-24397: Implement FieldDefinition form mapper for ezinteger

### PR DESCRIPTION
**Pull request by glye**
Manual test. Requires this change in the fieldvalue converter to work: https://github.com/ezsystems/ezpublish-kernel/pull/1264

https://jira.ez.no/browse/EZP-24397

### CODE REVIEW

**Comment by glye** [comment]
The doc says it uses IntegerValueValidator: https://doc.ez.no/display/EZP/The+Integer+FieldType
Dunno yet why it fails. Does it receive empty string, or string "0" maybe?

**Comment by glye** [comment]
Rebased after the ezfloat commit. Review ping @lolautruche

**Comment by glye** [comment]
Added 'empty_data' => 0 to min/max values, otherwise form complains of invalid value when submitting. Unsure if this is correct: How can you submit the default max value? (which is false, i.e. not a valid integer)

File: lib/FieldType/Mapper/IntegerFormMapper.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use Symfony\Component\Form\FormInterface;
+
+class IntegerFormMapper implements FieldTypeFormMapperInterface
+{
+    public function mapFieldDefinitionForm(FormInterface $fieldDefinitionForm, FieldDefinitionData $data)
+    {
+        $fieldDefinitionForm
+            ->add(
+                'minValue', 'integer', [
+                    'empty_data' => 0,
```
Comment (lolautruche) [inline lib/FieldType/Mapper/IntegerFormMapper.php]: What about negative values?

**Comment by glye** [comment]
Reverted 'empty_data' change, added better fix in https://github.com/ezsystems/ezpublish-kernel/pull/1264

**Comment by glye** [comment]
Rebased due to upstream id changes in translation file, no code change.

**Comment by pspanja** [comment]
+1

**Comment by glye** [comment]
@lolautruche About negative values, 0 is the default for minvalue, and setting 0 for empty_data afaik does not stop you from using negative values. Anyway, I removed it, so no longer relevant.

**Comment by lolautruche** [comment]
+1

**Pull request merged**